src/utils/getSigner.ts

```diff
@@ -0,0 +1,141 @@
+import type { ethers } from 'ethers';
+import { KJUR } from 'jsrsasign';
+import { createConnectWalletButton } from './CheckoutWithEth/createConnectWalletButton';
+import { createWalletConnectorContainer } from './CheckoutWithEth/createWalletConnectorContainer';
+import { importWagmiLibraries } from './CheckoutWithEth/importWagmi';
+import { isTestnet } from './CheckoutWithEth/isTestnet';
+import { mapWalletConnectorNameToPaperWalletName } from './CheckoutWithEth/mapWalletConnectorToPaperWalletName';
+import { setUpSigner } from './CheckoutWithEth/setUpSigner';
+
+export async function getSignerInfo({
+  container,
+  sdkClientSecret,
+  appName,
+}: {
+  container: HTMLElement;
+  sdkClientSecret: string;
+  appName?: string;
+}) {
+  try {
+    const {
+      fetchSigner,
+      configureChains,
+      connect,
+      createClient,
+      goerli,
+      mainnet,
+      MetaMaskConnector,
+      WalletConnectLegacyConnector,
+      CoinbaseWalletConnector,
+    } = await importWagmiLibraries();
+    const { alchemyProvider } = await import('@wagmi/core/providers/alchemy');
+
+    const alchemyApiKey = 'k5d8RoDGOyxZmVWy2UPNowQlqFoZM3TX';
+    const payloadObj = KJUR.jws.JWS.readSafeJSONString(
+      Buffer.from(sdkClientSecret.split('.')[1], 'base64').toString('utf-8'),
+    );
+    if (!payloadObj) {
+      throw new Error('Invalid sdkClientSecret given');
+    }
+    // TODO: Stop hacking types LoL
+    const chainName = (payloadObj as any).pricingDetails.chainName;
```

Comment:
Hmm, we're importing an entire library just to get whether to use testnet funds. Can we do it without the library?
https://stackoverflow.com/a/38552302

Reply:
oh ya you could, I was not thinking straight last night LoL